src/modules/ekf2/ekf2_main.cpp

```diff
@@ -860,17 +925,25 @@ void Ekf2::run()
 					    && airspeed.timestamp > 0) {
 						ctrl_state.airspeed = airspeed.indicated_airspeed_m_s;
 						ctrl_state.airspeed_valid = true;
+
+					} else {
+						ctrl_state.airspeed = sqrtf(v_n(0) * v_n(0) + v_n(1) * v_n(1) + v_n(2) * v_n(2));
+						ctrl_state.airspeed_valid = false;
```

Comment:
In these two cases we no longer know what the wind speed is, so the ground speed is used as a surrogate and the validity is set to false.